### PR TITLE
Devise

### PR DESCRIPTION
**Pull request by banzaman**
- Install and set up devise.
- Can register a new user.
- User logs in with a combination of email and password.
- Hashed passwords should be stored in the database.
- Ask for confirmation by email.
- Can reset password.

### CODE REVIEW

File: app/views/devise/registrations/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
+<% end %>
+
```
Comment (danmainah) [inline app/views/devise/registrations/new.html.erb:33]: - [x] @Team Great work on your registration page, however the layout slightly differed with the given design, the button width should be equal to the input fields width, check the comparison below

Wireframe Design          |  Your design     
:-------------------------:|:-------------------------:
image |  image

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -1,7 +1,10 @@
 class ApplicationController < ActionController::Base
-  helper_method :current_user
+  before_action :configure_permitted_parameters, if: :devise_controller?
```
Comment (danmainah) [inline app/controllers/application_controller.rb:2]: - [ ] @Team Great work setting up devise for your app, however you forgot to ensure users login in to the app are authenticated, currently I don't need to login to access any page, kindly consider adding user authentication in your app
```suggestion
  before_action :configure_permitted_parameters, if: :devise_controller?
  before_action :authenticate_user!
```

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @banzaman and @mohamedgamaleldin999999 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Setup Devise correctly
✔️  Ensure users are able to register and login successfully


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: app/views/devise/registrations/new.html.erb (hunk 1 above)
Comment (banzaman) [reply app/views/devise/registrations/new.html.erb:33]: Let's work on that

File: app/views/devise/registrations/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  </div>
+
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "new-password" %>
+  </div>
+
+  <div class="actions">
+    <%= f.submit "Sign up" %>
+  </div>
```
Comment (rloterh) [inline app/views/devise/registrations/new.html.erb:31]: - [x] Let's note that this activity requires the modification of the views in _devise_ for the registration/login to also matches the styling of the [provided wireframe(s)](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md). As mentioned in the previous review, let's ensure that the form (inputs and buttons) stretches across the screen and is centered accordingly for the **signup** page, as seen in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md):

**Your Signup/login view:**  
image

**Expected Wireframe Signup view:**
image

- [x] [OPTIONAL]
The position of the form make it difficult for users easily access the email input field. Let's also ensure that the login form is properly aligned/positioned on the login page,  as seen in the [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/sneak_peek.md)

**Email input not easily accessible:**
image

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Creed, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🏁 🚦 ⛪ 🚜 🏃 🌴 🏫 🌳 🏘️

## Highlights 👍
- Devise authentication parameters properly set!  
- Authentication is coming up well so far, good job!
- ReadMe looks great!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's improve the look of the ReadMe by removing the underlined aspect:

image

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by skyv26 — APPROVED** [review]
## Hi @banzaman and @mohamedgamaleldin999999, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## STATUS: APPROVED 🟢 



## Highlights
- Good **PR** title ✅ 
- Descriptive *PR* description ✅ 
- Implemented the features by adjusting the UI as per given design ✅ 

image

## Have any doubt ❓ 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @skyv26 in your question so I can receive the notification. You can also follow me on the below platforms

<table>
<tr>
<td>

<a href="https://github.com/skyv26">
   @skyv26
</a>

</td>
<td>

<a href="https://www.linkedin.com/in/skyv2022/">
   linkedin-skyv2022
</a>

</td>
<td>

<a href="https://twitter.com/vrma_aakash">
   @vrma_aakash
</a>

</td>
<td>

<a href="https://www.instagram.com/skybrel/">
   @skybrel
</a>

</td>
<td>

<a href="https://www.frontendmentor.io/profile/skyv26">
   @skybrel
</a>

</td>
</tr>
</table>

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by banzaman** [comment]
Thank you for your time

File: app/views/devise/registrations/new.html.erb (hunk 3 above)
Comment (banzaman) [reply app/views/devise/registrations/new.html.erb:31]: Let's work on that

**Pull request merged**